x-pack/legacy/plugins/integrations_manager/server/integrations/data.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  SavedObject,
+  SavedObjectsBulkGetObject,
+  SavedObjectsClientContract,
+} from 'src/core/server/saved_objects';
+import { InstallationSavedObject, Installable } from '../../common/types';
+import { SAVED_OBJECT_TYPE } from '../../common/constants';
+import * as Registry from '../registry';
+
+export async function getIntegrations(client: SavedObjectsClientContract) {
+  const registryItems = await Registry.fetchList();
+  const searchObjects: SavedObjectsBulkGetObject[] = registryItems.map(({ name, version }) => ({
+    type: SAVED_OBJECT_TYPE,
+    id: `${name}-${version}`,
+  }));
+
+  const results = await client.bulkGet(searchObjects);
+  const savedObjects: InstallationSavedObject[] = results.saved_objects.filter(o => !o.error); // ignore errors for now
```

Comment:
do you know if the saved object client takes a type parameter for what its methods return? I like being able to do:
```ts
const results = await client.bulkGet<InstallationSavedObject>(searchObjects);
```
and not having to explicitly type the return of the filter, if the method handles that properly. Not a blocker just curious.

Reply:
I haven’t seen that before but, yeah, they do. 

Or something quite similar https://github.com/elastic/kibana/blob/40a1bdee897c23c60980418c3aa958ee877ccb3d/src/core/server/saved_objects/service/saved_objects_client.ts#L339 & https://github.com/elastic/kibana/blob/40a1bdee897c23c60980418c3aa958ee877ccb3d/src/core/server/saved_objects/service/saved_objects_client.ts#L149-L164

Take a look at https://github.com/elastic/kibana/pull/40685/commits/98dfac89375b258a4219fe515745c5b8ae477602. I updated all the `client.*` calls and removed the explicit typing of their return values.